### ISSUE TITLE
gmsh.cc: add test that passes an input .dmg model

### ISSUE THREAD

**Issue by cwsmith** [SCOREC/core#415]
We need to that a case where the first argument, the input model, is not `none`:

https://github.com/SCOREC/core/blob/d7cc6c8341e8154a3e33f4746a044d4ebfd84178/test/testing.cmake#L231-L244

This caused problems with a downstream tool: https://github.com/SCOREC/gmshToPumi/issues/2

**Comment by zhangchonglin** [comment]
@cwsmith: thanks for creating this issue. From looking at the `gmsh.cc` code:
https://github.com/SCOREC/core/blob/d7cc6c8341e8154a3e33f4746a044d4ebfd84178/test/gmsh.cc#L40-L52

- it seems that with `gmsh version 4`, the `.dmg` file is not needed, where function `loadMdsDmgFromGmsh` does not accept `.dmg` input file:
https://github.com/SCOREC/core/blob/d7cc6c8341e8154a3e33f4746a044d4ebfd84178/mds/mdsGmsh.cc#L469-L475

- for `gmsh version 2`, `.dmg` input file is an optional file.

It seems that the `if statement` corresponding to `gmshVersion == 4` needs to be restructured to be consistent with all types of input arguments.

**Comment by cwsmith** [comment]
> it seems that with gmsh version 4, the .dmg file is not needed
That is correct, v4 can embed sufficient info into the `.msh` file for creating the output `.dmg`.  I'm not sure if *all* v4 gmsh meshes will have this info though.

> loadMdsDmgFromGmsh does not accept .dmg input file
Yeah, thanks for digging.  

Regarding the if statement in `gmsh.cc` for v4,  I agree, there should at least be an `else` that prints a helpful error message.
